client/app/bundles/course/survey/containers/ResponseForm/index.jsx

```diff
@@ -75,50 +74,48 @@ export const buildResponsePayload = (data) => {
   return { response: { answers_attributes, submit: data.submit } };
 };
 
-class ResponseForm extends Component {
-  static renderSections(props) {
-    const { fields, disabled } = props;
-    return (
-      <>
-        {fields.map((member, index) => {
-          const section = fields.get(index);
-          return (
-            <ResponseSection
-              key={section.id}
-              {...{ member, index, fields, disabled }}
-            />
-          );
-        })}
-      </>
-    );
-  }
-
-  componentDidMount() {
-    window.addEventListener('beforeunload', this.handleUnload);
-  }
+const ResponseForm = (props) => {
+  const {
+    initialValues,
+    onSubmit,
+    readOnly,
+    response,
+    flags: { canSubmit, canModify, isResponseCreator },
+  } = props;
+  const {
+    control,
+    handleSubmit,
+    reset,
+    setError,
+    formState: { errors, isSubmitting, isDirty },
+  } = useForm({
+    defaultValues: initialValues,
+  });
+  const { fields } = useFieldArray({
+    control,
+    name: 'sections',
+  });
 
-  componentWillUnmount() {
-    window.removeEventListener('beforeunload', this.handleUnload);
-  }
+  useEffect(() => {
+    reset(initialValues);
+  }, [initialValues]);
```

Comment:
I think `reset` should also be a dependency, not sure.

Reply:
Yep my concern is that initial values do change, but the `reset` function being called is no longer the "latest" one anymore.

Reply:
I think this will still call the latest reset. Just that if the reset changes, we dont want the form to be reset when the initialValues are not changed.